core/include/traccc/clusterization/detail/sparse_ccl.hpp

```diff
@@ -111,27 +111,22 @@ TRACCC_HOST_DEVICE inline unsigned int sparse_ccl(
     for (unsigned int i = 0; i < n_cells; ++i) {
         L[i] = i;
         unsigned int ai = i;
-        if (i > 0) {
-            for (unsigned int j = start_j; j < i; ++j) {
-                if (is_adjacent(cells[i], cells[j])) {
-                    ai = make_union(L, ai, find_root(L, j));
-                } else if (is_far_enough(cells[i], cells[j])) {
-                    ++start_j;
-                }
+        for (unsigned int j = start_j; j < i; ++j) {
+            if (is_adjacent(cells[i], cells[j])) {
+                ai = make_union(L, ai, find_root(L, j));
+            } else if (is_far_enough(cells[i], cells[j])) {
+                ++start_j;
             }
         }
     }
 
     // second scan: transitive closure
     for (unsigned int i = 0; i < n_cells; ++i) {
-        unsigned int l = 0;
         if (L[i] == i) {
-            l = labels;
-            ++labels;
+            L[i] = labels++;
         } else {
-            l = L[L[i]];
+            L[i] = L[L[i]];
         }
-        L[i] = l;
```

Comment:
hmm is this correct change?

Reply:
Ah I think it is right.

L[i] will still be `l`which is before ++ operator, right?

Reply:
yes, that's correct, `i++` returns the value of `i` before incrementing, contrarily to `++i`.
https://en.cppreference.com/w/cpp/language/operator_incdec